builtin/common/tests/vector_spec.lua

```diff
@@ -25,6 +25,31 @@ describe("vector", function()
 		end)
 	end)
 
+	it("indexes", function()
+		local some_vector = vector.new(24, 42, 13)
+		assert.equal(24, some_vector[1])
+		assert.equal(24, some_vector.x)
+		assert.equal(42, some_vector[2])
+		assert.equal(42, some_vector.y)
+		assert.equal(13, some_vector[3])
+		assert.equal(13, some_vector.z)
+
+		some_vector[1] = 100
+		assert.equal(100, some_vector[1])
```

Comment:
maybe test the opposite here: `some_vector.x`

Reply:
Done.